File: src/database/DBTransportUnit.java
package database;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;

import model.*;

/**
 * DBTransportUnit
 * Handles all database functionality for default transport units.
 *
 * @author Dani Sander
 * @version 1.0
 * @since 20-05-15
 */

public class DBTransportUnit {
    
    private DBConnection dbConnection;
    private static DBTransportUnit instance;
    
    /**
     * Private constructor for singleton.
     * @throws SQLException 
     * @throws ClassNotFoundException 
     */
    private DBTransportUnit() throws ClassNotFoundException, SQLException {
        dbConnection = DBConnection.getInstance();        
    }
    
    /**
     * Singleton method for class.
     * @return instance of class.
     * @throws SQLException 
     * @throws ClassNotFoundException 
     */
    public static DBTransportUnit getInstance() throws ClassNotFoundException, SQLException {
        if (instance == null) {
            instance = new DBTransportUnit();            
        }
        
        return instance;
    }

    /**
     *
     * @param IDs
     * @return
     */
    public ArrayList<TransportUnit> getTransportUnits(ArrayList<Long> IDs) {
        ArrayList<TransportUnit> list = new ArrayList<>();

        IDs.stream().forEach((ID) -> {
            ArrayList<TransportUnit> tmp_list;
            String sql = String.format("select * from Customer where customer_id = '%s';", ID);
            tmp_list = (ArrayList<TransportUnit>) dbConnection.sendSQL(this, sql, "_formatTransportUnit");

            tmp_list.forEach((tmp) -> list.add(tmp));
        });

        return list;
    }

    /**
     * @param rs takes the ResultSet from database
     * @return list of TransportUnit
     */
    public ArrayList<TransportUnit> _formatTransportUnit(ResultSet rs) {
        ArrayList<TransportUnit> tableList = new ArrayList<>();
        try {
            while (rs.next()) {
                tableList.add(
                        new TransportUnit(
                                rs.getLong("id"),
                                rs.getLong("customer_id"),
                                Type.valueOf(rs.getString("type"))
                        ));
            }
        } catch (SQLException e) {
            // TODO Auto-generated catch block
            e.printStackTrace();
        }
        return tableList;
    }
}

File: Server/Database/DBTransportUnit.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Model;
using System.Data;

namespace Server.Database
{
    public class DBTransportUnit
    {
        private DBConnection DbConnection { get; set; }
        private static DBTransportUnit instance;

        /// private constructor for singelton     
        private DBTransportUnit()
        {
            DbConnection = DBConnection.Instance;
        }

        /// singelton get instance method 
        /// returns the instance from DB 
        public static DBTransportUnit Instance
        {
            get
            {
                if (instance == null)
                {
                    instance = new DBTransportUnit();
                }

                return instance;
            }

        }

        /// <summary>
        /// Returns a list of all TransportUnits.
        /// </summary>
        /// <param name="defaultDeliveryStopID"></param>
        /// <returns></returns>
        public List<TransportUnit> getTransportUnits(List<long> IDs)
        {
            List<TransportUnit> list = new List<TransportUnit>();

            foreach (long id in IDs) 
            {
                List<TransportUnit> tmp_list;
                String sql = String.Format("select * from TransportUnit where customer_id = '{0}';", id);
                tmp_list = DbConnection.SendSQL<TransportUnit>(sql, ConvertToTransportUnit);
                foreach (TransportUnit t in tmp_list) {
                    list.Add(t);
                }
            }
            return list;
        }


        /// <summary>
        /// Takes the ReulstSet from database and returns a list of all TransportUnits.
        /// </summary>
        /// <param name="rs"></param>
        /// <returns></returns>
        public List<TransportUnit> ConvertToTransportUnit(IDataReader dataSet) 
        {
            List<TransportUnit> tableList = new List<TransportUnit>();
            while (dataSet.Read()) {
                tableList.Add(new TransportUnit(
                    dataSet.GetInt64(0),
                    dataSet.GetInt64(1),
                    dataSet.GetDouble(2)
                ));
            }
            
            return tableList;
        }
    }
}